numba/cpython/randomimpl.py

```diff
@@ -1713,13 +1720,11 @@ def poisson_impl2(lam, size):
        (isinstance(size, types.UniTuple) and isinstance(size.dtype,
                                                         types.Integer))
     ):
-        def _impl(lam, size):
-            out = np.empty(size, dtype=np.intp)
-            out_flat = out.flat
-            for idx in range(out.size):
-                out_flat[idx] = np.random.poisson(lam)
-            return out
         return _impl
+    if isinstance(lam, (types.Float, types.Integer)) and (
+        isinstance(size, types.Tuple) and isinstance(size.dtype, types.Integer)
```

Comment:
I am very skeptical about this code path. The `types.Tuple` is a heterogeneous tuple, so that means `size.dtype` will always be `types.UnionType` -- so is this effectively dead code?